modules/test/playwright/tests/portal-workflow-kaleo-designer-web/assignments.spec.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {workflowPagesTest} from '../../fixtures/workflowPagesTest';
+import {getRandomInt} from '../../utils/getRandomInt';
+
+export const test = mergeTests(apiHelpersTest, loginTest(), workflowPagesTest);
+
+let workflowDefinitionId: number;
+let workflowDefinitionName: string;
+
+test.beforeEach(async ({apiHelpers}) => {
+	const singleApproverWorkflowDefinition =
+		await apiHelpers.headlessAdminWorkflow.getWorkflowDefinitionByName(
+			'Single Approver'
+		);
+
+	workflowDefinitionName = 'Copy of Single Approver' + getRandomInt();
+
+	const workflowDefinition =
+		await apiHelpers.headlessAdminWorkflow.postWorkflowDefinitionSave(
+			workflowDefinitionName,
+			singleApproverWorkflowDefinition
+		);
+
+	workflowDefinitionId = workflowDefinition.id;
+});
+
+test.afterEach(async ({apiHelpers, scriptManagementPage}) => {
+	await apiHelpers.headlessAdminWorkflow.deleteWorkflowDefinition(
+		workflowDefinitionId
+	);
+
+	await scriptManagementPage.enableScriptManagementConfiguration();
+});
+
+test('can see Scripted Assignment option when script management configuration is enabled', async ({
+	diagramViewPage,
+	nodePropertiesSidebarPage,
+	page,
+	processBuilderPage,
+}) => {
+	await processBuilderPage.goto();
+
+	await processBuilderPage.clickWorkflowDefinitionName(
+		workflowDefinitionName
+	);
+
+	await diagramViewPage.clickUpdateNodeLink();
+
+	await nodePropertiesSidebarPage.editAssignmentButton.click();
+
+	expect(
+		await page.$(`#assignment-type option[value="scriptedAssignment"]`)
```

Comment:
The use of element handles is [discouraged by Playwright](https://playwright.dev/docs/api/class-elementhandle). Have you tried using Locators?